sklearn/linear_model/_base.py

```diff
@@ -267,7 +284,8 @@ def _decision_function(self, X):
         check_is_fitted(self)
 
         X = self._validate_data(X, accept_sparse=["csr", "csc", "coo"], reset=False)
-        return safe_sparse_dot(X, self.coef_.T, dense_output=True) + self.intercept_
+        coef_T = self.coef_ if (self.coef_.ndim < 2) else self.coef_.T
+        return X @ coef_T + self.intercept_
```

Comment:
Similarly, I find the following more straightforward:

```suggestion
        if self.coef_.ndim == 2:
            return X @ self.coef_.T + self.intercept_
        else:
            return X @ self.coef_ + self.intercept_
```

Reply:
Done (except I conditioned on `coef_.ndim == 1` rather than on `coef.ndim == 2`)